Jetpack E2E: add Social Connection spec using Tumblr.

worldomonation opened the pull request:
Related to https://github.com/Automattic/wp-calypso/issues/80730.

## Proposed Changes

This PR adds the piping, secrets and the actual E2E to test the Social Connection.

Key changes:
- add new category in the secrets file, for social media connections. Accounts for these are in the secret store.
- new methods in RestAPIClient to interact with the publicize endpoint.
- new E2E test to connect Tumblr account to the site.

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Jetpack E2E Simple (mobile)
  - [x] Jetpack E2E Simple (desktop)
  - [x] Jetpack E2E AT (desktop)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/secrets/types.ts
packages/calypso-e2e/src/secrets/secrets-manager.ts
packages/calypso-e2e/src/secrets/encrypted.enc
- update secrets with new social connections category

packages/calypso-e2e/src/lib/pages/marketing-page.ts
- add methods to handle social connections

packages/calypso-e2e/src/types/rest-api-client.types.ts
- new methods to get and delete publicize connections

test/e2e/specs/tools/social-connections__tumblr.ts
- add new spec to connect and disconnect social connection

dpasque commented on test/e2e/specs/tools/social-connections__tumblr.ts:
```diff
@@ -0,0 +1,79 @@
+		);
+
+		// If it does, remove the connection.
+		Object.values( connections ).forEach( async ( connection ) => {
+			if ( connection.label === 'Tumblr' ) {
+				console.info( `Removing existing connection for Tumblr for accountName ${ accountName }.` );
+				await restAPIClient.deletePublicizeConnection( connection.site_ID, connection.ID );
+			}
+		} );
```
**Suggestion (blocking):** As a heads up, it's usually a bad idea to mix `forEach` with `async-await`! It doesn't work how you might expect. I just got burned by this yesterday haha. 😆 For this specific case, you could do a `for... of` on the `connections`.

[Helpful reference](https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971)

worldomonation replied:
Thanks for the heads up! I always get confused on how best to iterate over an array and perform async action on each in JS/TS.

dpasque commented on packages/calypso-e2e/src/lib/pages/marketing-page.ts:
```diff
@@ -66,4 +83,69 @@ export class MarketingPage {
+	 *
+	 * @param {SocialConnection} target Social service.
+	 */
+	async validateSocialConnected( target: SocialConnection ): Promise< boolean > {
+		try {
+			await this.page
+				.getByRole( 'main' )
+				.getByRole( 'listitem' )
+				.filter( { hasText: target } )
+				.getByRole( 'button', { name: 'Disconnect' } )
+				.waitFor();
+			return true;
+		} catch {
+			return false;
+		}
+	}
```
**Nit (non-blocking):** In our previous test design patterns, [and even in our docs](https://wpcalypso.wordpress.com/devdocs/test/e2e/docs/patterns_tricks_gotchas.md), `validate` functions throw instead of returning booleans, so I think that might be more consistent to do here if you're okay with it!

worldomonation replied:
Good reminder, thank you!

dpasque reviewed (approved):
Great job! 🥳 Two suggestions, one blocking, but I don't need to review again! 🚀

worldomonation committed:
packages/calypso-e2e/src/lib/pages/marketing-page.ts
- change return signature of validateSocialConnected method.

worldomonation committed:
test/e2e/specs/tools/social-connections__tumblr.ts
- replace Object.forEach with for...of loops

worldomonation committed:
test/e2e/specs/tools/social-connections__tumblr.ts
- replace sidebar navigation with direct visits to the page.

packages/calypso-e2e/src/lib/pages/marketing-page.ts
- add a `waitForLoadState` wait for the popup.

worldomonation committed:
packages/calypso-e2e/src/rest-api-client.ts
- add JSDoc

worldomonation committed:
test/e2e/specs/tools/social-connections__tumblr.ts
- add exception for private sites
